File: c#/task1/14mod1.cs
using System; // Аналог <iostream>

class Program { // Весь код всегда внутри класса, точка входа - main или т. п. 
    static bool Towers(string str, int x){
        int n = str.Length;  // Длина строки
        bool[] cover = new bool[n];
        for (int i = 0; i < n; i++){
            if (str[i] == '1') 
                cover[i] = true;  // Если на этой позиции есть вышка, она покрывает участок
            else 
                cover[i] = false;  // Если вышки нет, участок не покрыт
        }
        for (int i = 0; i < n; i++){
            if (cover[i]){  // Если на этой позиции есть вышка
                for (int k = 0; k < x; k++){
                    if (i + k < n) cover[i + k] = true;  // Участок справа от вышки
                    if (i - k >= 0) cover[i - k] = true;  // Участок слева от вышки
                }
            }
        }
        for (int i = 0; i < n; i++){
            if (!cover[i])  
                return false;
        }

        return true;  
    }

    static void Main(){
        Console.WriteLine("Введите вышки");
        string s = Console.ReadLine();  
        Console.WriteLine("Введите показатель x");
        int x = int.Parse(Console.ReadLine());  

        if (Towers(s, x)) 
            Console.WriteLine("Yes");
        else 
            Console.WriteLine("No");
    }
}

File: rust/task1/14mod1.rs
use std::io::{self, Write};

fn towers(str: &str, x: usize) -> bool {
    let n = str.len(); // Получаем длину строки
    let mut cover = vec![false; n]; // Вектор длины n, все элементы — false, mul - изменяемая переменная

    // Преобразуем строку в список символов
    for (i, c) in str.chars().enumerate() { // str.chars().enumerate() - преобразует строку в символы и индексирует их
        if c == '1' {
            cover[i] = true; // Отмечаем, где стоят вышки
        }
    }

    for i in 0..n {
        if cover[i] {
            for k in 0..=x {
                if i + k < n {
                    cover[i + k] = true; // Покрытие вправо
                }
                if i >= k {
                    cover[i - k] = true; // Покрытие влево
                }
            }
        }
    }

    // Проверяем, все ли участки покрыты
    for c in cover {
        if !c {
            return false; // Если хотя бы один не покрыт — возвращаем false
        }
    }

    true // Иначе возвращаем true
}

fn main() {
    let mut input = String::new(); // Считываем ввод пользователя со строки

    println!("Введите вышки: ");
    io::stdout().flush().unwrap(); // Очищаем буфер вывода 
    io::stdin().read_line(&mut input).unwrap(); // Читаем строку с клавиатуры
    let s = input.trim().to_string(); // Удаляем пробелы и перевод строки

    // Очищаем строку перед следующим вводом
    input.clear();

    // Просим пользователя ввести значение X
    println!("Введите показатель X: ");
    io::stdout().flush().unwrap();
    io::stdin().read_line(&mut input).unwrap();

    // Преобразуем строку в число
    let x: usize = match input.trim().parse() {
        Ok(num) => num, // Успешное преобразование
        Err(_) => {
            println!("Строка должна быть целым числом");
            return;
        }
    };


    if towers(&s, x) {
        println!("Yes");
    } else {
        println!("No");
    }
}
